src/javaharness/java/arcs/api/PortablePromiseFactory.java

```diff
@@ -0,0 +1,11 @@
+package arcs.api;
+
+import java.lang.Runnable;
+import java.util.function.Consumer;
+
+/**
+ * Portable Promise factory that works on either JS or Android environment.
+ */
+public interface PortablePromiseFactory {
+  <T> PortablePromise<T> newPromise(Consumer<T> resolve, Runnable reject);
```

Comment:
I think there is a confusion here. When you call the Promise() constructor is JS, you pass in a function that takes two arguments. e.g. new Promise(function(resolve, reject) { /* do work and call resolve() or reject() */ })

These functions are promised by the underlying Promise machinery, not you. Here I think they are being conflated with user provided functions which you intend to be called *when* a rejection or a resolution happens. 

See here:  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise#Syntax

There are normally defined by doing Promise.then(resolvedFunc).catch(rejectedFunc)

I think what you need instead is something like this:

interface PortablePromiseExecutor<T> {
   void doInvoke(Consumer<T> resolve, Runnable reject);
}

Then people would do this:
promiseFactory.newPromise((resolve, reject) -> {
   resolve.accept(value);
   // else reject.run()
});